cmd/curio/guidedsetup/guidedsetup.go

```diff
@@ -0,0 +1,593 @@
+	if err != nil {
+		d.say(notice, "Error reading from database: %s. Aborting Migration.", err.Error())
+		os.Exit(1)
+	}
+
+	if os.Getenv("FULLNODE_API_INFO") == "" {
+		for _, binName := range []string{"lotus", "./lotus"} {
+			cmd := exec.Command(binName, "net", "listen")
+			b := bytes.NewBuffer(nil)
+			cmd.Stdout = b
+			if err := cmd.Run(); err != nil {
+				continue
+			}
+			for _, addr := range strings.Split(b.String(), "\n") {
+				os.Setenv("FULLNODE_API_INFO", addr)
+				break
+			}
+		}
+	}
+	{
+		var closer jsonrpc.ClientCloser
+
+		d.full, closer, err = cliutil.GetFullNodeAPI(d.cctx)
+		if err != nil {
+			path := os.Getenv("LOTUS_PATH")
+			if path == "" {
+				path = must.One(homedir.Expand("~/.lotus"))
+			}
+			ainfos, err2 := cliutil.GetAPIInfoFromRepoPath(path, repo.FullNode)
+			if err2 != nil {
+				d.say(notice, "Error connecting to lotus node: %s %s", err.Error(), err2.Error())
+				os.Exit(1)
+			}
+			// connect to apiinfo
+			for _, ainfo := range ainfos {
+				addr, err := ainfo.DialArgs("v0")
+				if err != nil {
+					continue
+				}
+				d.full, closer, err = client.NewFullNodeRPCV0(context.Background(), addr, ainfo.AuthHeader())
+				if err == nil {
+					break
+				}
+			}
+		}
+		d.closers = append(d.closers, closer)
+	}
+	ainfo, err := cliutil.GetAPIInfo(d.cctx, repo.FullNode)
+	if err != nil {
+		d.say(notice, "could not get API info for FullNode: %w", err)
+		os.Exit(1)
+	}
+	token, err := d.full.AuthNew(context.Background(), api.AllPermissions)
+	if err != nil {
+		d.say(notice, "Error getting token: %s", err.Error())
+		os.Exit(1)
+	}
+
+	chainApiInfo := fmt.Sprintf("%s:%s", string(token), ainfo.Addr)
+
+	d.MinerID, err = SaveConfigToLayer(d.MinerConfigPath, "", false, chainApiInfo)
+	if err != nil {
+		d.say(notice, "Error saving config to layer: %s. Aborting Migration", err.Error())
+		os.Exit(1)
+	}
+}
+
+// bucket returns the power's 4 highest bits (rounded down).
+func bucket(power *api.MinerPower) uint64 {
+	rawQAP := power.TotalPower.QualityAdjPower.Uint64()
+	leadingDigit := 64 - bits.LeadingZeros64(rawQAP)
+	if leadingDigit < 4 {
+		return 0
+	}
+	return rawQAP >> (uint64(leadingDigit) - 4) << (uint64(leadingDigit - 4))
+}
+
+func oneLastThing(d *MigrationData) {
+	d.say(section, "Protocol Labs wants to improve the software you use. Tell the team you're using Curio.")
+	i, _, err := (&promptui.Select{
+		Label: d.T("Select what you want to share with the Curio team."),
+		Items: []string{
+			d.T("Individual Data: Miner ID, Curio version, net (%s or %s). Signed.", "mainnet", "testnet"),
+			d.T("Aggregate-Anonymous: version, net, and Miner power (bucketed)."),
+			d.T("Hint: I am someone running Curio on net."),
+			d.T("Nothing.")},
+		Templates: d.selectTemplates,
+	}).Run()
+	if err != nil {
+		d.say(notice, "Aborting remaining steps.", err.Error())
+		os.Exit(1)
+	}
+	if i < 3 {
+		msgMap := map[string]any{
+			"chain": build.BuildTypeString(),
```

Comment:
I'm going to keep the message key to our server as "net" since the database roundtrip is tested for that and it's only visible to code readers who will not likely be confused. In any case, this code will go away in less than a year.

Reply:
ok @magik6k ?